sdk/state/state.go

```diff
@@ -43,6 +43,8 @@ type Channel struct {
 
 	latestCloseAgreement     CloseAgreement
 	latestUnconfirmedPayment Payment
+
+	coordinatedClose CoordinatedClose
```

Comment:
Coordinating a close is really just modifying the previous close agreement. I think it would be helpful because of that if we store the signatures for it in a CloseAgreement in the latestCloseAgreement. This way the last close is still there in the same place, it can just be executed early.

I think this would require adding the observation periods to the close agreement, and so merging the CoordinatedClose and CloseAgreement types.

Wdyt?

Reply:
But there would be two kinds of closeSignatures in that case right? eg:
```
type CloseAgreement struct {
	IterationNumber       int64
	Balance               Amount
	CloseSignatures       []xdr.DecoratedSignature
	DeclarationSignatures []xdr.DecoratedSignature

        // new:
	CoordinatedCloseSignature []xdr.DecoratedSignature
}
```
That felt awkward to me when I tried that, so created another type for separation of concerns.  🤔

Reply:
Hmm, good point. We'd also be mutating an object that conceptually was locked in previously.

I think the thing I'm stuck on is that the channel has only two states we should be tracking:
- The most recent fully signed agreement.
- The agreement we're considering right now, and that agreement is either a new agreement (a new payment) of the adjustment of some attribute of the most recent agreement.

If the `lastUnconfirmedAgreement` was a `CloseAgreement` rather than a `Payment`, and if the `CloseAgreement` stored the observation periods then that would probably work. 

Storing them separately works too, but I worry we've now go to check in a few places if a coordinate close or a payment is in progress, because if either is in progress we shouldn't let the other begin, that could be dangerous.

Reply:
> Storing them separately works too, but I worry we've now go to check in a few places if a coordinate close or a payment is in progress, because if either is in progress we shouldn't let the other begin, that could be dangerous.

that's true. I'm changing the close code to work without a `CoordinatedClose` type to see how it looks. I'll have a PR out